asserts/snapasserts/validation_sets.go

```diff
@@ -27,20 +27,67 @@ import (
 
 	"github.com/snapcore/snapd/asserts"
 	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/naming"
 )
 
+// InstalledSnap holds the minimal details about an installed snap required to
+// check the validation set.
+type InstalledSnap struct {
+	naming.SnapRef
+	Revision snap.Revision
+}
+
+// NewInstalledSnap creates InstalledSnap.
+func NewInstalledSnap(name, snapID string, revision snap.Revision) *InstalledSnap {
+	return &InstalledSnap{
+		SnapRef:  naming.NewSnapRef(name, snapID),
+		Revision: revision,
+	}
+}
+
+func (e *ValidationSetsConflictError) Error() string {
```

Comment:
this seems to have been moved to a weird place it should be close to the error def